### PR TITLE
[CLOUD-563] Move grants to DML

### PR DESCRIPTION
**Pull request by maismail**
Fixes https://hopsworks.atlassian.net/browse/CLOUD-563

### CODE REVIEW

File: templates/default/sql/dml/3.2.0.sql.erb [hunk 1]
```diff
@@ -6,4 +6,28 @@ REPLACE INTO `hopsworks`.`variables`(`id`, `value`) VALUES ("git_image_version",
...
+GRANT NDB_STORED_USER ON *.* TO '<%= node['hopsworks']['mysql']['user'] %>'@'127.0.0.1';
+GRANT ALL PRIVILEGES ON <%= node['hopsworks']['db'] %>.* TO '<%= node['hopsworks']['mysql']['user']%>'@'127.0.0.1';
+GRANT SELECT ON <%= node['hops']['db'] %>.* TO '<%= node['hopsworks']['mysql']['user']%>'@'127.0.0.1';
+GRANT ALL PRIVILEGES ON <%= node['hops']['db'] %>.yarn_projects_quota TO '<%= node['hopsworks']['mysql']['user'] %>'@'127.0.0.1';
+GRANT ALL PRIVILEGES ON <%= node['hops']['db'] %>.hdfs_directory_with_quota_feature TO '<%= node['hopsworks']['mysql']['user'] %>'@'127.0.0.1';
+GRANT SELECT ON metastore.* TO '<%= node['hopsworks']['mysql']['user'] %>'@'127.0.0.1';
+
+-- Grants for Kafka 
+CREATE USER IF NOT EXISTS '<%= node['kkafka']['mysql']['user'] %>'@'127.0.0.1' IDENTIFIED BY '<%= node['kkafka']['mysql']['password'] %>';
+REVOKE ALL PRIVILEGES ON *.* FROM '<%= node['kkafka']['mysql']['user'] %>'@'127.0.0.1';
```
Comment (kouzant) [inline templates/default/sql/dml/3.2.0.sql.erb:23]: This should to go to the update, right?

**Review by kouzant — CHANGES_REQUESTED** [review]

File: templates/default/sql/dml/3.2.0.sql.erb (hunk 1 above)
Comment (maismail) [reply templates/default/sql/dml/3.2.0.sql.erb:23]: I guess you mean upgrade? If so, yes we always run all the DML files on upgrade

Comment (kouzant) [reply templates/default/sql/dml/3.2.0.sql.erb:24]: Yeah sorry my bad, I was confused

File: recipes/default.rb [hunk 2]
```diff
@@ -291,7 +291,8 @@
          :onlinefs_password => encrypted_onlinefs_password,
          :onlinefs_salt => onlinefs_salt,
          :pki_ca_configuration => caConf.to_json(),
-         :usernames_configuration => usernamesConfiguration.to_json()
+         :usernames_configuration => usernamesConfiguration.to_json(),
+         :kafka_installed => exists_local("kkafka", "default")
```
Comment (kouzant) [inline recipes/default.rb:295]: You should have the `valid_recipe` as before because Kafka might not be on the same host as Hopsworks

**Review by kouzant — CHANGES_REQUESTED** [review]

**Review by kouzant — APPROVED** [review]

**Pull request merged**